allocator.go

```diff
@@ -184,3 +218,66 @@ func (a *Allocator) Allocate() (ok bool, assignments map[RangeID][]NodeID) {
 	}
 	return true, res
 }
+
+// Minimize the difference absolute difference from the previous assignment and the current assignment
+func (a *Allocator) adhereToMinimalChurnConstraint() {
+
+	if !a.opts.churnOptions.WithMinimalChurn && !a.opts.churnOptions.withMaxChurn {
+		return
+	}
+
+	var vars = make([]cpsatsolver.IntVar, 0, 2*len(a.ranges)*len(a.nodes))
+	var coefficients = make([]int64, 0, len(a.ranges)*len(a.nodes))
+	o := toMap(a.opts.churnOptions.prevAssignment)
```

Comment:
These helpers are going to go away, but as an aside -- these are bad names. What's `o`?